public/app/core/components/Select/UserPicker.test.tsx

```diff
@@ -1,14 +1,17 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { act, render, screen } from '@testing-library/react';
 import { UserPicker } from './UserPicker';
 
 jest.mock('@grafana/runtime', () => ({
   getBackendSrv: () => ({ get: jest.fn().mockResolvedValue([]) }),
 }));
 
 describe('UserPicker', () => {
-  it('renders correctly', () => {
-    render(<UserPicker onSelected={() => {}} />);
+  it('renders correctly', async () => {
+    await act(async () => {
```

Comment:
I'm not entirely sure if this is necessary since RTL will wrap the component automatically with `act` 🤔 

https://github.com/testing-library/react-testing-library/blob/8f17a2bc4fa9c2c17a623b38ded9ab18b086e457/src/pure.js#L59-L65

Reply:
It does but if component has a state, which is set async, that wrapper is not enough and such components will still emit the warning.

Reply:
Because I wanted to clarify this myself:

```javascript
function Component() {
  const [state, setState] = useState();
  useEffect(() => setState("hello"), []);
  return <div>component: {state}</div>;
}
```

RTL's render function's internal act will catch this state update correctly, and will not require us to wrap + await act(). This is because the state update happens immediately within `useEffect`, which will be immediately "flushed" by `act()`

However, once we make the useEffect state update async, the `act()` flush will no longer catch it

```javascript
function AsyncStateComponent() {
  const [state, setState] = useState();
  useEffect(() => {
    Promise.resolve().then(() => setState("hello"));
  }, []);
  return <div>component: {state}</div>;
}
```